fe/fe-core/src/main/java/com/starrocks/catalog/PrimitiveType.java

```diff
@@ -69,598 +70,95 @@
-        builder.put(VARCHAR, TIME);
-        // DecimalV2
-        builder.put(DECIMALV2, BOOLEAN);
-        builder.put(DECIMALV2, TINYINT);
-        builder.put(DECIMALV2, SMALLINT);
-        builder.put(DECIMALV2, INT);
-        builder.put(DECIMALV2, BIGINT);
-        builder.put(DECIMALV2, LARGEINT);
-        builder.put(DECIMALV2, FLOAT);
-        builder.put(DECIMALV2, DOUBLE);
-        builder.put(DECIMALV2, DECIMALV2);
-        builder.put(DECIMALV2, CHAR);
-        builder.put(DECIMALV2, VARCHAR);
-        builder.put(DECIMALV2, DECIMAL32);
-        builder.put(DECIMALV2, DECIMAL64);
-        builder.put(DECIMALV2, DECIMAL128);
-        builder.put(DECIMALV2, TIME);
-
-        // Decimal32
-        builder.put(DECIMAL32, BOOLEAN);
-        builder.put(DECIMAL32, TINYINT);
-        builder.put(DECIMAL32, SMALLINT);
-        builder.put(DECIMAL32, INT);
-        builder.put(DECIMAL32, BIGINT);
-        builder.put(DECIMAL32, LARGEINT);
-        builder.put(DECIMAL32, FLOAT);
-        builder.put(DECIMAL32, DOUBLE);
-        builder.put(DECIMAL32, DECIMALV2);
-        builder.put(DECIMAL32, CHAR);
-        builder.put(DECIMAL32, VARCHAR);
-        builder.put(DECIMAL32, DECIMAL32);
-        builder.put(DECIMAL32, DECIMAL64);
-        builder.put(DECIMAL32, DECIMAL128);
-        builder.put(DECIMAL32, TIME);
-
-        // Decimal64
-        builder.put(DECIMAL64, BOOLEAN);
-        builder.put(DECIMAL64, TINYINT);
-        builder.put(DECIMAL64, SMALLINT);
-        builder.put(DECIMAL64, INT);
-        builder.put(DECIMAL64, BIGINT);
-        builder.put(DECIMAL64, LARGEINT);
-        builder.put(DECIMAL64, FLOAT);
-        builder.put(DECIMAL64, DOUBLE);
-        builder.put(DECIMAL64, DECIMALV2);
-        builder.put(DECIMAL64, CHAR);
-        builder.put(DECIMAL64, VARCHAR);
-        builder.put(DECIMAL64, DECIMAL32);
-        builder.put(DECIMAL64, DECIMAL64);
-        builder.put(DECIMAL64, DECIMAL128);
-        builder.put(DECIMAL64, TIME);
-
-        // Decimal128
-        builder.put(DECIMAL128, BOOLEAN);
-        builder.put(DECIMAL128, TINYINT);
-        builder.put(DECIMAL128, SMALLINT);
-        builder.put(DECIMAL128, INT);
-        builder.put(DECIMAL128, BIGINT);
-        builder.put(DECIMAL128, LARGEINT);
-        builder.put(DECIMAL128, FLOAT);
-        builder.put(DECIMAL128, DOUBLE);
-        builder.put(DECIMAL128, DECIMALV2);
-        builder.put(DECIMAL128, CHAR);
-        builder.put(DECIMAL128, VARCHAR);
-        builder.put(DECIMAL128, DECIMAL32);
-        builder.put(DECIMAL128, DECIMAL64);
-        builder.put(DECIMAL128, DECIMAL128);
-        builder.put(DECIMAL128, TIME);
-
-        // HLL
-        builder.put(HLL, HLL);
+    private static final ImmutableList<PrimitiveType> INTEGER_TYPE_LIST =
+            ImmutableList.of(TINYINT, SMALLINT, INT, BIGINT, LARGEINT);
 
-        // BITMAP
-        builder.put(BITMAP, BITMAP);
+    private static final ImmutableList<PrimitiveType> FLOAT_TYPE_LIST =
```

Comment:
Could combine INTEGER_TYPE_LIST and FLOAT_TYPE_LIST to NUMBER_TYPE_LIST?

Reply:
Maybe, these lists needs to be further refactored, so we could reuse the list. Then we need to differ INTEGER_TYPE and FLOATPOINT_TYPE. 
But I could build another NUMBER_TYPE_LIST to combine both of them all.